### PR TITLE
Add app ID to state

### PR DESCRIPTION
**Pull request by Cloud11PL**
* Exposes app ID in state (that is passed as a query param) - used by https://github.com/saleor/saleor-app-marketplace/pull/19

### CODE REVIEW

**Review by jwm0 — APPROVED** [review]

File: src/index.ts [hunk 1]
```diff
@@ -12,7 +12,9 @@ export function createApp(targetDomain?: string) {
     domain = url.searchParams.get("domain") || "";
   }
 
-  app.setState({ domain });
+  id = new URL(domain).searchParams.get("id") ?? "";
```
Comment (dominik-zeglen) [inline src/index.ts:15]: Nitpick but it's not changed anywhere

```suggestion
  const id = new URL(domain).searchParams.get("id") ?? "";
```

File: src/types.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import createApp from ".";
+import { AppBridgeState } from "./app";
+
+export type App = ReturnType<typeof createApp>;
```
Comment (dominik-zeglen) [inline src/types.ts:4]: Not a fan of this, explicitly typing return type by `function createApp(): Type` is making breaking api harder. Now it's possible to change function signature and no one will know until consumers start complaining about breaking changes.

**Review by dominik-zeglen — APPROVED** [review]

File: test/index.test.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ Object.defineProperty(window.document, "referrer", {
 import { actions, DispatchResponseEvent, createApp } from "../src";
 
 describe("createApp", () => {
-  const domain = "test-domain";
+  const domain = "http://test-domain.com/?id=appid";
```
Comment (magul) [inline test/index.test.ts:13]: in dashboard we have `shop.domain.host` passed here, isn't that just a domain (without schema)? also as you pass ``src={urlJoin(src, `?domain=${shop.domain.host}&id=${appId}`)}`` in dashboard, should that be `test-domain.com&id=appid` here?

File: src/index.ts [hunk 4]
```diff
@@ -12,7 +12,9 @@ export function createApp(targetDomain?: string) {
     domain = url.searchParams.get("domain") || "";
   }
 
-  app.setState({ domain });
+  const id = new URL(domain).searchParams.get("id") ?? "";
```
Comment (magul) [inline src/index.ts:15]: instead of that should we have `id = url.searchParams.get("id")` in `else` clause above?

**Pull request merged**